lib/index.js

```diff
@@ -77,13 +87,46 @@ export default {
     this.subscriptions.dispose();
   },
 
+  fixJob(editor) {
+    let correctScope = false;
+
+    if (editor && editor.buffer && editor.buffer.file && editor.buffer.file.path) {
+      const { path } = editor.buffer.file;
+
+      this.baseScopes.forEach((scope) => {
+        const extension = scope.replace('source', '');
+
+        if (path.lastIndexOf(extension) === path.length - extension.length) {
+          correctScope = true;
+        }
+      });
+    }
+
+    if (!correctScope) {
+      return;
+    }
+
+    const oldFileContent = editor.getText();
+    const cursorPosition = editor.getCursorBufferPosition();
+
+    this.provideLinter().lint(editor, true).then(() => {
```

Comment:
This will actually initialize a _new_ provider, I'm not sure how problematic that would be, but maybe `provideLinter` should cached the object in `this` (so package activation/deactivation still gets a new instance)?